compatibility_spark-3/src/test/scala/za/co/absa/hyperdrive/compatibility/impl/writer/delta/DeltaTestBase.scala

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright 2018 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package za.co.absa.hyperdrive.compatibility.impl.writer.delta
+
+import org.apache.spark.sql.execution.streaming.MemoryStream
+import org.scalatest.{BeforeAndAfterEach, FlatSpec}
+import za.co.absa.commons.io.TempDirectory
+import za.co.absa.hyperdrive.shared.utils.SparkTestBase
+
+trait DeltaTestBase extends BeforeAndAfterEach with SparkTestBase {
+  this: FlatSpec =>
+
+  val baseDir: TempDirectory = TempDirectory("DeltaTempDir").deleteOnExit()
+  val baseDirPath: String = baseDir.path.toAbsolutePath.toString
+  val destinationPath = s"${baseDir.path.toAbsolutePath.toString}/destination"
+  val checkpointPath = s"${baseDir.path.toAbsolutePath.toString}/checkpoint"
+
+  import spark.implicits._
+
+  val memoryStream: MemoryStream[CDCEvent] = MemoryStream[CDCEvent](1, spark.sqlContext)
+
+  override def beforeEach(): Unit = {
+    import org.apache.commons.io.FileUtils
+    FileUtils.cleanDirectory(baseDir.path.toAbsolutePath.toFile)
```

Comment:
Do you really need this? I think you could also delete the directory afterEach and recreate it beforeEach. I don't like importing a "random" dependency, just for a minor convenience.

Reply:
Done